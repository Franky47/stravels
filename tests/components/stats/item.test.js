import 'react-native'
import React from 'react'
import StatItem from '@stravels/components/stats/item'
// import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

test('component renders correctly', () => {
  const tree = renderer.create(<StatItem label='foo' value={42} unit='bar' />).toJSON()
  expect(tree).toMatchSnapshot()
})

// test('onPress', () => {
//   let i = 0 // i guess i could have used sinon here too... less is more i guess
//   const onPress = () => i++
//   const wrapperPress = shallow(<FullButton onPress={onPress} text='hi' />)

//   expect(wrapperPress.prop('onPress')).toBe(onPress) // uses the right handler
//   expect(i).toBe(0)
//   wrapperPress.simulate('press')
//   expect(i).toBe(1)
// })
