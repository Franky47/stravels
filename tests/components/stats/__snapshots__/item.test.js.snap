// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontFamily": "Avenir-Book",
          "fontSize": 24,
        }
      }
    >
      42
    </Text>
     
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#aaa",
          "fontSize": 11,
          "fontWeight": "700",
        }
      }
    >
      BAR
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#aaa",
        "fontSize": 11,
        "fontWeight": "500",
      }
    }
  >
    FOO
  </Text>
</View>
`;
